### PR TITLE
chore(ci): notify slack when build result changes

### PR DESCRIPTION
**Pull request by pihme**
Changes the Jenkinsfile to notify Slack whenever the build result changes.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -204,16 +204,17 @@ pipeline {
...
                 }
-
-                slackSend(
-                    channel: '#zeebe-testbench-ci',
-                    message: "Zeebe Cluster Testbench failed on ${env.BRANCH_NAME} for build ${currentBuild.absoluteUrl}"
-            )
+                if (hasBuildResultChanged()) {
+                    slackSend(
+                        channel: '#zeebe-testbench-ci',
+                        message: "Zeebe Cluster Testbench failed on ${env.BRANCH_NAME} for build ${currentBuild.absoluteUrl}"
```
Comment (korthout) [inline Jenkinsfile]: Don't forget to change the message 🙂

**Review by korthout — APPROVED** [review]
Looks good, just one thing, but here's the approval already 👍

File: Jenkinsfile (hunk 1 above)
Comment (pihme) [reply Jenkinsfile]: Good point

**Pull request merged**